From: Gabriel Krisman Bertazi <krisman@suse.de>
Date: Fri, 24 Jun 2026 12:50:10 -0400
Subject: kabi: io_uring: Handle time_namespace header include
Patch-mainline: Never, kabi workaround
References: bsc#1271291

patches.suse/io_uring-wait-honour-caller-s-time-namespace-for-IOR.patch
adds an include of linux/time_namespace.h to io_uring.c, changing the
CRC of symbols that assumed a forward declaration, but not risking any
problems.  Add the usual GENKSYMS pre-processor check.

Signed-off-by: Gabriel Krisman Bertazi <krisman@suse.de>
Reviewed-by: Michal Koutný <mkoutny@suse.com>

--- io_uring/io_uring.c | 2 ++
1 file changed, 2 insertions(+)

--- a/io_uring/io_uring.c
+++ b/io_uring/io_uring.c
@@ -67,7 +67,9 @@
 #include <linux/fadvise.h>
 #include <linux/task_work.h>
 #include <linux/io_uring.h>
+#ifndef __GENKSYMS__
 #include <linux/time_namespace.h>
+#endif
 #include <linux/io_uring/cmd.h>
 #include <linux/audit.h>
 #include <linux/security.h>
